sci-libs/scikits_learn/scikits_learn-0.22.1.ebuild

```diff
@@ -0,0 +1,103 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit distutils-r1 flag-o-matic
+
+MYPN="${PN/scikits_/scikit-}"
+MYP="${MYPN}-${PV}"
+
+DESCRIPTION="Python modules for machine learning and data mining"
+HOMEPAGE="https://scikit-learn.org"
+SRC_URI="mirror://pypi/${MYPN:0:1}/${MYPN}/${MYP}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc examples test"
+RESTRICT="!test? ( test )" 
+
+# tried to unbundle virtual/python-funcsigs, funcsigs, odict
+# but it is a large mess to maintain
+
+RDEPEND="
+	dev-python/matplotlib[${PYTHON_USEDEP}]
+	dev-python/nose[${PYTHON_USEDEP}]
+	dev-python/numpy[lapack,${PYTHON_USEDEP}]
+	sci-libs/scikits[${PYTHON_USEDEP}]
+	sci-libs/scipy[${PYTHON_USEDEP}]
+	virtual/blas:=
+	virtual/cblas:=
+"
+
+DEPEND="
+	dev-python/cython[${PYTHON_USEDEP}]
+	dev-python/numpy[lapack,${PYTHON_USEDEP}]
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	sci-libs/scipy[${PYTHON_USEDEP}]
+	virtual/blas:=
+	virtual/cblas:=
+"
+
+S="${WORKDIR}/${MYP}"
+
+python_prepare_all() {
+	# bug #397605
+	[[ ${CHOST} == *-darwin* ]] \
+		&& append-ldflags -bundle "-undefined dynamic_lookup" \
+		|| append-ldflags -shared
+
+	# scikits-learn now uses the horrible numpy.distutils automagic
+	export SCIPY_FCONFIG="config_fc --noopt --noarch"
+
+	# remove bundled cblas
+	rm -rf sklearn/src || die "failed to remove bundled cblas"
+
+	# commented out, since it is a mess to maintain
+	# use system joblib
+	#rm -r sklearn/externals/joblib || die
+	#sed -i -e '/joblib/d' sklearn/externals/setup.py || die
+	#for f in sklearn/{*/,}*.py; do
+	#	sed -r -e '/^from/s/(sklearn|\.|)\.externals\.joblib/joblib/' \
+	#		-e 's/from (sklearn|\.|)\.externals import/import/' -i $f || die
+	#done
+
+	# use system funcsigs and odict
+	#rm sklearn/externals/funcsigs.py || die
+	#rm sklearn/externals/odict.py || die
+	#for f in sklearn/{utils/fixes.py,gaussian_process/{tests/test_,}kernels.py}; do
+	#	sed -r -e 's/from (sklearn|\.|)\.externals\.funcsigs/from funcsigs/' -i $f || die
+	#done
+	distutils-r1_python_prepare_all
+}
+
+python_compile() {
+	distutils-r1_python_compile ${SCIPY_FCONFIG}
+}
+
+python_test() {
+	distutils_install_for_testing ${SCIPY_FCONFIG}
+	esetup.py install \
```

Comment:
Why do you need to install it twice?

Reply:
I'm still confused as to why you need to do that. After all, this shouldn't make any difference compared to the contents of `${BUILD_DIR}/lib`.

Reply:
Might have something to do with the parameters. In any case, `distutils_install_for_testing` fails with this [build.log](https://github.com/gentoo/gentoo/files/4392273/build.log).

Reply:
What about without any install command at all, just using build_dir?

Reply:
What do you mean by “just using `build_dir`”?

Reply:
`${BUILD_DIR}/lib` to pytest

Reply:
No, that won't work at all, as it doesn't even find most of the tests. The closest I could get it to actually working is by using:

```
    #esetup.py install \
    #   --root="${T}/test-${EPYTHON}"
    #pushd "${T}/test-${EPYTHON}/$(python_get_sitedir)" || die > /dev/null
    #pytest -m "not network" -vv || die
    #popd > /dev/null
    distutils_install_for_testing
    ${EPYTHON} -m pytest -v "${BUILD_DIR}/test/lib/sklearn" || die
```

This fails with the following [build.log](https://github.com/gentoo/gentoo/files/4398081/build.log).

If I comment out the new version and uncomment the commented lines, it works. I would suggest we go with what works.

Reply:
@mgorny so, is it ok if I just go with:

```
    esetup.py install \
       --root="${T}/test-${EPYTHON}"
    pushd "${T}/test-${EPYTHON}/$(python_get_sitedir)" || die > /dev/null
    pytest -m "not network" -vv || die
    popd > /dev/null
```

That's about as standard as I could get it while still having it work.